src/Sarif.Multitool.Library/Rules/SARIF2007.ExpressPathsRelativeToRepoRoot.cs

```diff
@@ -58,7 +58,10 @@ protected override void Analyze(Run run, string runPointer)
                     }
                     else
                     {
-                        this.uriBaseIds.Add(run.VersionControlProvenance[i].MappedTo.UriBaseId);
+                        string validUriBaseId = run.VersionControlProvenance[i].MappedTo.UriBaseId;
+                        this.uriBaseIds.Add(validUriBaseId);
+                        run.OriginalUriBaseIds?.Where(baseId => baseId.Value?.UriBaseId == validUriBaseId).ToList()
+                            .ForEach(baseId => this.uriBaseIds.Add(baseId.Key));
```

Comment:
I think you could replace this for:
```
if (run.OriginalUriBaseIds != null)
{
    foreach (var baseId in run.OriginalUriBaseIds(..)
    {
         this.uribaseids.add(baseid.key)
    }
}
```

with that, you won't need to call toList and generate everything at first.